### PR TITLE
Add CA categories page

### LINKED ISSUE #22457: Subpages for `Naming` and `Formatting` rules should be moved to their parent node
Also, the subpages for Naming rules and Formatting rules should just be moved to their parent node in the TOC.

_Originally posted by @gewarren in https://github.com/dotnet/docs/issues/22196#issuecomment-754086087_

---

@gewarren Please feel free to add more details (for example: what is meant by *parent node*).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b0a396c1-9c21-75a9-70b5-799cafc941b5
* Version Independent ID: 6d91b560-c12f-6e6d-e92b-1babb27edb67
* Content: [Code style rules overview - .NET](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/)
* Content Source: [docs/fundamentals/code-analysis/style-rules/index.md](https://github.com/dotnet/docs/blob/master/docs/fundamentals/code-analysis/style-rules/index.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
- Add a new category index page
- Link to the various ways knowing the category is useful
- Other miscellaneous code analysis improvements
- Fixes #22457, contributes to #22678.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/categories?branch=pr-en-us-22673).

### CODE REVIEW

File: docs/fundamentals/code-analysis/categories.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+| [Maintainability rules](quality-rules/maintainability-warnings.md) | Maintainability rules support library and application maintenance. |
+| [Naming rules](quality-rules/naming-warnings.md) | Naming rules support adherence to the naming conventions of the .NET design guidelines. |
+| [Performance rules](quality-rules/performance-warnings.md) | Performance rules support high-performance libraries and applications. |
+| [Publish rules](quality-rules/publish-warnings.md) | Publish rules support single-file applications. |
+| [Reliability rules](quality-rules/reliability-warnings.md) | Reliability rules support library and application reliability, such as correct memory and thread usage. |
+| [Security rules](quality-rules/security-warnings.md) | Security rules support safer libraries and applications. These rules help prevent security flaws in your program. |
+| [Style rules](style-rules/index.md) | Style rules support consistent code style in your codebase. These rules start with the "IDE" prefix. |
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/categories.md:27]: I included style rules as a category. @mavasani please confirm this is correct.

Comment (mavasani) [reply docs/fundamentals/code-analysis/categories.md:27]: Yes, this looks good.

**Review by mavasani — APPROVED** [review]
Looks great!

**Comment by gewarren** [comment]
Thanks Manish!

**Pull request merged**